src/main/java/com/amazon/ion/impl/bin/AbstractIonWriter.java

```diff
@@ -87,90 +87,124 @@ public final void writeValue(final IonReader reader) throws IOException
 
     public final void writeValueRecursive(final IonReader reader) throws IOException
```

Comment:
Perhaps it would be good to add a comment saying that it does a depth-first (recursion-like) traversal of the all values on the IonReader, but the method is not actually implemented using recursion.

Reply:
Added